### PR TITLE
Added github actions

### PR DESCRIPTION
**Pull request by AlessandroMinoccheri**

### CODE REVIEW

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container: pugx/poser:php80
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Validate composer.json and composer.lock
```
Comment (JellyBellyDev) [inline .github/workflows/php.yml:14]: here we don't have composer.lock versioned

File: .github/workflows/php.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          config: phpspec.yml
+
+      - name: Run tests (phpspec)
+        run: ./bin/phpspec run --format=pretty
+
+      - name: Run coverage tests (phpspec)
+        run: ./bin/phpspec run -f progress -c phpspec-coverage.yml
+
+      - name: Run behat tests
+        run: ./bin/behat
```
Comment (JellyBellyDev) [inline .github/workflows/php.yml:49]: maybe the jobs for php74 are missing?

**Review by JellyBellyDev — CHANGES_REQUESTED** [review]

**Comment by JellyBellyDev** [comment]
why i don't see `actions` in the checks?

File: .github/workflows/php.yml (hunk 1 above)
Comment (AlessandroMinoccheri) [reply .github/workflows/php.yml:14]: yes I can remove composer.lock

File: .github/workflows/php.yml (hunk 2 above)
Comment (AlessandroMinoccheri) [reply .github/workflows/php.yml:49]: Yes, I am thinking on what is the best approach but I think that the correct way is to use a PHP version matrix

**Comment by AlessandroMinoccheri** [comment]
Because there is a problem inside docker images, asap I change our docker images to docker github actions images.

File: .github/workflows/php.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        run: composer validate
+
+      - name: Cache Composer packages
+        id: composer-cache
+        uses: actions/cache@v2
+        with:
+          path: vendor
+          key: ${{ runner.os }}-php-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-php-
```
Comment (JellyBellyDev) [inline .github/workflows/php.yml:29]: `${{ runner.os }}-php-${{ matrix.container }}-${{ hashFiles('**/composer.lock') }}`
and
`${{ runner.os }}-php-${{ matrix.container }}`
WDYT?

File: .github/workflows/php.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+on:
+  push:
+  pull_request:
+
+jobs:
+  build:
+
+    strategy:
+      matrix:
+        container: [ "php74  ", "php80" ]
```
Comment (JellyBellyDev) [inline .github/workflows/php.yml:12]: typo on `space`

**Review by JellyBellyDev — CHANGES_REQUESTED** [review]

**Review by JellyBellyDev — APPROVED** [review]
LGTM! 👍

**Pull request merged**